packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1496,6 +1504,204 @@ class SliverAppBar extends StatefulWidget {
        assert(stretchTriggerOffset > 0.0),
        assert(collapsedHeight == null || collapsedHeight >= toolbarHeight, 'The "collapsedHeight" argument has to be larger than or equal to [toolbarHeight].');
 
+  /// Creates a Material Design medium top app bar that can be placed
+  /// in a [CustomScrollView].
+  ///
+  /// Returns a [SliverAppBar] configured with appropriate defaults
+  /// for a medium top app bar as defined in Material 3. It starts fully
+  /// expanded with the title in an area underneath the main row of icons.
+  /// When the [CustomScrollView] is scrolled, the title will be scrolled
+  /// under the main row. When it is fully collapsed, a smaller version of the
+  /// title will fade in on the main row. The reverse will happen if it is
+  /// expanded again.
+  ///
+  /// {@tool dartpad}
+  /// This sample shows how to use [SliverAppBar.medium] in a [CustomScrollView].
+  ///
+  /// ** See code in examples/api/lib/material/app_bar/sliver_app_bar.2.dart **
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  ///   * [AppBar], for a small or center-aligned top app bar.
+  ///   * [SliverAppBar.large], for a large top app bar.
+  ///   * https://m3.material.io/components/top-app-bar/overview, the Material 3
+  ///     app bar specification.
+  factory SliverAppBar.medium({
```

Comment:
Bless you for omitting deprecated parameters. ❤️ 

Is `forceElevated` intentionally left out?

Reply:
> Is `forceElevated` intentionally left out?

Somewhat, as it is kind of redundant now that we have two elevations (`elevation` & `scrolledUnderElevation`). But for people that don't want to specify both, and to be consistent with the other APIs perhaps I should add it.